memorydatabase.go

```diff
@@ -0,0 +1,84 @@
+package naffka
+
+import (
+	"fmt"
+	"sync"
+)
+
+// A MemoryDatabase stores the message history as arrays in memory.
+// It can be used to run unit tests.
+// If the process is stopped then any messages that haven't been
+// processed by a consumer are lost forever.
+type MemoryDatabase struct {
+	topicsMutex sync.Mutex
+	topics      map[string]*memoryDatabaseTopic
+}
+
+type memoryDatabaseTopic struct {
+	messagesMutex sync.Mutex
+	messages      []Message
+}
+
+func (t *memoryDatabaseTopic) addMessage(msg *Message) {
+	t.messagesMutex.Lock()
+	defer t.messagesMutex.Unlock()
+	for int64(len(t.messages)) < msg.Offset {
+		t.messages = append(t.messages, Message{})
```

Comment:
Interesting that we pad out with stub messages up to the offset. I'm assuming this API is documented somewhere, can you add a link as a line comment please.